### DISCUSSION TITLE
Add Cloudlog integration

### DISCUSSION THREAD

**Discussion by artekw** [foldynl/QLog#340] [Ideas]
It is posible to add export QSO to Cloudlog ?

**Comment by foldynl** [comment]
Simply, yes, it is possible but question is. Should this be implemented? 

Offline sending of messages can be implemented immediately. On the other hand, I don't see any possibility to implement online sending of message to Cloudlog, because Cloudlog has not implemented the Update/Delete operations. So if you edit or delete QSO in QLog, it will not be updated in Cloudlog.

Another very important question is whether Cloudlog should be included in the QLog menu. If so, where is the boundary of which "online" services should be added and which should not. Over time the QLog "Services" menu can expand to several pages.

Currently, you can use QLog's [notifications](https://github.com/foldynl/QLog/wiki/Notifications) to send messages to Cloudlog online - if someone implements a bridge between Qlog and Cloudlog. But as I wrote, the problem starts when you try to edit/delete QSO in QLog.

It would be great to have a synergy between QLog and Cloudlog for synchronization, but everything is limited by Cloudlog's [API](https://github.com/magicbug/Cloudlog/wiki/API).

**Comment by artekw** [comment]
You are right about Cloudlog API limitations.
But what you think about add some button "Export to Cloudlog" or similar ?

**Comment by foldynl** [comment]
All ends with a button. It's not about whether it will be a button or menu options; it's about whether the function fits into the concept. Let's take a look: LoTW, ClubLog, QRZ.com, eQSL, and HRDLog services are now implemented. These services offer the following features:

1. They provide additional services to operators, which QLog essentially cannot offer (such as QSL services, QSO confirmation, issuing diplomas, etc.).
2. They are supported by ADIF format. This is crucial for exporting/importing QSOs between applications without losing information about whether the  QSO has already been sent to the online service or not = QSO Sent status.

I consider these two points crucial for a service to be integrated into QLog. Comparatively, Cloudlog unfortunately does not provide such services. Actually, it is the second application with the same functionality as QLog provides. Consequently, exporting QSOs from one application to another is what you are looking for and QLog already offers this function.

Unfortunately, the Cloudlog's APIs are so simple that it cannot even be said that QLog would gain anything by implementing an interface for Cloudlog. It's a shame that there is no possibility of a "synchronization API". In this case, I believe that QLog would be the first to implement it, because I have many requests for synchronization.

Therefore, I think that at the moment I do not see a sufficient reason to implement it in QLog. Maybe I'm not seeing something what I should know.

**Comment by artekw** [comment]
Thanks for the clarification. All that remains is to "synchronize" manually until the Cloudlog API is improved :-)

**Reply by diliak** [reply]
Hello,
Yes, i was working on this problem for hours before find this topic !
My solution it to suppress contact on wavelog and reupload it with Qlog...
It should be good that wavelog accept to "modify QSO"...
Wait and see...

**Comment by foldynl** [comment]
sorry, additional question for my curiosity. I would like to understand it more. Why should you use two apps that do the same thing? The only difference is online/offline logging. If I have an online log, I don't need offline applications. I would only need this in special cases when I don't have internet and as far as I know, Cloudlog community is already talking about the possibility of offline logging. So why to have both applications?

**Comment by artekw** [comment]
I find QLog logging friendly, it has everything on one screen (last qso, cluster dx, etc.). But I want to be able to access my journal when I'm away from QTH. Cloudlog is also a backup. All I need is a "dumb" "Export to Cloudlog" button like Log4OM, but I find this program very slow. 
I will try to perform such export on my own using notifications.

**Reply by diliak** [reply]
Hello,
Yes, same for my use here. I consider Wavelog as a basic logbook only for backup and web site use (i can put my log on read only accessible by public).
But the problem is that Wavelog devs consider that their script is "MASTER" logbook.

The possible solution is that in Qlog/Service/Upload QSO's, it was possible to select BASIC transmitting or something else without QSL's working... Actually, The Qlog tool see if there is an LOTW update, for example.

Other solution (perhaps simplest solution) is no updating QSO's (transmit NEW one ONLY). So, if we want to update QSO's on Wavelog, we will use WAVELOG. If we consider that Wavelog is MASTER logbook, it is logical.

**Reply by diliak** [reply]
I reply to myself : 
I tested other thing. I edited one QSO on QLOG, changed nothing and tested Service again. One new QSO to upload on QRZ, CLUBLOG, HRDLOG and WAVELOG.
Yes, it is logical if we change something.

It seems that there is already no impact when QSL's updates on Qlog.
The problem is really on Wavelog script that must accept "MODIFY" or "REPLACE" QSO,s.
Will see the script or submit it to us.

**Comment by foldynl** [comment]
Cloudlog support is part of version 0.45.

**Comment by oliverbross** [comment]
Cloudlog implementation works well, but I do suggest one fix. When contacts are uploaded to Cloudlog, they show both rx/tx frequencies which looks messy, here is the comparision how it should be and how imported or synced contacts looks like:

<img width="392" height="179" alt="image" src="https://github.com/user-attachments/assets/d6bb977f-2f07-47e9-9d5b-4bfb298979c9" />

The suggestions is simple, do not add addif field for TX if the contact was made simplex. Only use it for split contacts.

Also, this causes issues on other systems / logs as the RX/TX freq are the same and it looks very messy. This is not needed. 

Otherwise uploads are working 100% fine.

**Reply by foldynl** [reply]
> Cloudlog implementation works well, but I do suggest one fix. When contacts are uploaded to Cloudlog, they show both rx/tx frequencies which looks messy, here is the comparision how it should be and how imported or synced contacts looks like:
> <img alt="image" width="392" height="179" src="https://private-user-images.githubusercontent.com/44015391/470154974-d6bb977f-2f07-47e9-9d5b-4bfb298979c9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTMzNTAyNzEsIm5iZiI6MTc1MzM0OTk3MSwicGF0aCI6Ii80NDAxNTM5MS80NzAxNTQ5NzQtZDZiYjk3N2YtMmYwNy00N2U5LTlkNWItNGJmYjI5ODk3OWM5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzI0VDA5MzkzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAxMjU0NWNhZjJjYzE0YjBlYTZmNWY4ZTI3N2NkZDY3ZGQ0N2U1NDI1NmM3MDMwZWU2MDU3OWViNGFlNzYwNDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.VyZ1Pe3mw4d9n6cz_hhqnvJyxyIvs0ebcVfjSO-FfIU">
> 
> The suggestions is simple, do not add addif field for TX if the contact was made simplex. Only use it for split contacts.
> 
> Also, this causes issues on other systems / logs as the RX/TX freq are the same and it looks very messy. This is not needed.
> 
> Otherwise uploads are working 100% fine.

The question is whether this is rather a request for Wave/Cloudlog support. QLog is not the only logger that exports both RX and TX frequency when they are the same – for example, N1MM does this as well. Therefore, it seems that Cloud/Wavelog should be prepared for this and display it in a way that suits the user. 

QLog is not responsible for how the data is displayed in Cloud/Wavelog. QLog is responsible for providing the information in the required format, which in this case is ADIF. I recommend opening this as an issue in the Wave/Cloudlog project.

**Reply by oliverbross** [reply]
Thanks for your reply. I’ll raise this topic with the CloudLog/WaveLog teams as well. However, after testing multiple logging applications (including RumLog, KLog, MacLoggerDX, SwissLog, LogHX3, Log4OM, and others), I’ve observed that when exporting ADIF from these programs and importing into WaveLog/CloudLog, the frequency data appears correctly—showing a single frequency (RX) rather than separate RX/TX fields.

This issue seems specific to QLog’s ADIF export behavior. It appears QLog includes the TX frequency field in the ADIF export even when it is identical to the RX frequency (unless split mode is active). In contrast, most other logging software only include the FREQ_TX field in ADIF if:

The TX frequency differs from RX (e.g., split operation), or

The user explicitly enables split mode.

**Suggested Fix for QLog:**

Modify the ADIF export logic to omit FREQ_TX when:

FREQ_TX == FREQ_RX (or is blank/unset), and

Split mode is not active.

This aligns with industry-standard behavior (e.g., RumLog, Log4OM) and avoids unnecessary field duplication.

This should be a straightforward implementation since it only requires a conditional check before writing the FREQ_TX field to the ADIF output. Let me know if you’d like examples of ADIF exports from other software for comparison.
